src/beaverdam/viewer/dash_view.py

```diff
@@ -317,9 +343,63 @@ def filter_data(
                         "properties"
                     ]["style"]
                     if triggered_element_style == "pie":
+                        # Get the clicked value
                         new_filter_criteria = [
                             ctx.triggered[0]["value"]["points"][0]["label"]
                         ]
+
+                        # An idiosyncracy of Plotly pie charts is that when they are
+                        # created from booleans they convert the booleans to lowercase
+                        # strings ("true", "false", or "null" if the element is empty).
+                        # This doesn't happen with other plot types, e.g. bar chart.  As
+                        # a dirty workaround for this problem, if a user clicks on a pie
+                        # chart segment labelled with a boolean-like string, check the
+                        # types of values in the plotted data and make a guess about
+                        # which values to filter on.
+                        #
+                        # Define the possibilities for true and false values.  Because
+                        # odML values will be in lists, also allow lists of these
+                        # values.
+                        true_values = ["true", "True", "TRUE", True, 1, "1"]
+                        true_values.extend([[x] for x in true_values])
+                        false_values = ["false", "False", "FALSE", False, 0, "0"]
+                        false_values.extend([[x] for x in false_values])
+                        # Get ID of the plot
+                        plot_id = ctx.triggered[0]["prop_id"].split('"index":"')[1]
+                        plot_id = plot_id.split('"')[0]
+                        if new_filter_criteria in true_values:
+                            # Check which value is in the plotted values
+                            is_value_in_data = [
+                                self._is_in_plot_data(plot_id, x) for x in true_values
+                            ]
+                            # Take the first item that's in the plotted values, because
+                            # if e.g. the actual value is True, this also matches 1.
+                            # NOTE that this depends on the order that the truthy and
+                            # falsey values are put into the vectors!!
+                            actual_value = [
+                                true_values[ind]
+                                for ind, x in enumerate(is_value_in_data)
+                                if x
+                            ]
+                            new_filter_criteria = actual_value[0]
```

Comment:
Not sure I understand the argument for taking only the first value. Maybe this could be a bit more explicit in the comment on lines 375-378.